addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -1705,3 +1705,213 @@ def test04_rereservation_for_same_batch(self):
+                                       assign=True,
+                                       batch_id=batch.id)
+        batch.state = 'in_progress'
+        batch.reserve_pallet('UDES11111', picking=picking1)
+        picking2 = self.create_picking(self.picking_type_pick,
+                                       products_info=self.pack_4bananas_info,
+                                       confirm=True,
+                                       assign=True)
+        batch.reserve_pallet('UDES11112', picking=picking2)
+        self.assertFalse(picking2.u_reserved_pallet)
+
+    def test08_raises_exception_if_no_picking_provided(self):
+        batch = self.create_batch(user=self.outbound_user)
+        self.create_picking(self.picking_type_pick,
+                            products_info=self.pack_4apples_info,
+                            confirm=True,
+                            assign=True,
+                            batch_id=batch.id)
+        batch.state = 'in_progress'
+        expected_error = 'A picking must be specified if pallets are reserved per picking.'
+
+        with self.assertRaisesRegex(ValidationError, expected_error,
+                                    msg='Incorrect error thrown'):
+            batch.reserve_pallet('UDES11111')
+
+    def test09_raises_exception_if_maximum_reservable_pallets_exceeded(self):
+        self.picking_type_pick.u_max_reservable_pallets = 1
+        batch = self.create_batch(user=self.outbound_user)
+        picking = self.create_picking(self.picking_type_pick,
+                                      products_info=self.pack_4apples_info,
+                                      confirm=True,
+                                      assign=True,
+                                      batch_id=batch.id)
+        batch.state = 'in_progress'
+        batch.reserve_pallet('UDES11111', picking=picking)
+
+        self.create_picking(self.picking_type_pick,
+                            products_info=self.pack_4bananas_info,
+                            confirm=True,
+                            assign=True)
+        expected_error = 'Only %d pallets may be reserved at a time.' % self.picking_type_pick.u_max_reservable_pallets
+
+        with self.assertRaisesRegex(ValidationError, expected_error,
+                                    msg='Incorrect error thrown'):
+            batch.add_extra_pickings(self.picking_type_pick.id)
+
+    def test10_raises_exception_if_batch_created_with_more_than_max_reservable_pallets(self):
+        Batch = self.env['stock.picking.batch']
+
+        self.picking_type_pick.u_max_reservable_pallets = 1
+        pickings = self.create_picking(self.picking_type_pick,
+                                       products_info=self.pack_4apples_info,
+                                       confirm=True,
+                                       assign=True,
+                                       priority='0')
+        picking2 = self.create_picking(self.picking_type_pick,
```

Comment:
If we don't care about  individual pickings, we can use:
`pickings |=  XXX` 
then we don't need line 1909 and we can use pickings.ids in line 1917.